pkg/acs/calacs/acsccd/blev_funcs_postsm4.c

```diff
@@ -36,152 +40,243 @@ static int make_amp_array(const int arr_rows, const int arr_cols, SingleGroup *
 static int unmake_amp_array(const int arr_rows, const int arr_cols, SingleGroup * im,
                             int amp, double * array);
 
+/* Remove the signal dependent bias shift from post-SM4 full frame WFC data
+ * (based on ISR http://www.stsci.edu/hst/acs/documents/isrs/isr1202.pdf),
+ * as well as from supported subarray data (~post-mid-2016) as identified by
+ * specific aperture names 
+ * (based on ISR http://www.stsci.edu/hst/acs/documents/isrs/isr1703.pdf).
+ *
+ * chip2 consists of amps C & D, chip1 consists of amps A & B. 
+ *
+ * Because full frame data uses both chips, but supported subarray data
+ * is only from a single amp (therefore, a portion of a single chip), this
+ * routine has a variable number of input SingleGroups.
+ *
+ * *** This routine should be modified according to issues discussed in PR #312 and noted in IT #334. ***
+ *
+ * 16-May-2018 M.D. De La Pena: Generalized bias_shift_corr() to handle subarray data.
+ */
+int bias_shift_corr(ACSInfo *acs, int nGroups, ...) {
+
+  /* This function has built in assumptions regarding the data being passed */
+  assert (acs != NULL);  /* Make sure the ACSInfo structure is defined */
+  assert ((nGroups == 1) || (nGroups == 2)); /* Must be at least one and no more than two SingleGroups */
+  assert (((nGroups == 1) && (acs->subarray == YES)) || ((nGroups == 2) && (acs->subarray == NO)));  /* Subarray has one and full frame has two SingleGroups */
```

Comment:
Just so I've said it - this and the line above are ugly. The variadic arg should have been removed instead and these asserts handled by the compiler (at compile time).